### PR TITLE
Ensure the stream is closed immediately when a server response is received but still sending the request body

### PR DESCRIPTION
**Pull request by violetagg**

### CODE REVIEW

**Review by pderop — APPROVED** [review]
LGTM

**Review by simonbasle — CHANGES_REQUESTED** [review]
ok for the stream additional observer 👍 
one small typo in the constant name

File: reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientConfig.java [hunk 1]
```diff
@@ -935,4 +934,21 @@ public void onUncaughtException(Connection connection, Throwable error) {
 			}
 		}
 	}
+
+	static final class StreamConnectionObserver implements ConnectionObserver {
+
+		static final StreamConnectionObserver INSTANSE = new StreamConnectionObserver();
```
[the remark this answers is not in the document]
Comment (violetagg) [reply reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientConfig.java]: ouch ... fixed

**Review by simonbasle — APPROVED** [review]

**Comment by violetagg** [comment]
@pderop @simonbasle Thanks for the review!

**Pull request merged**